supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/statuspolling/JesApiQueryManager.scala

```diff
@@ -0,0 +1,105 @@
+package cromwell.backend.impl.jes.statuspolling
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props, SupervisorStrategy, Terminated}
+import cats.data.NonEmptyList
+import cromwell.backend.impl.jes.Run
+import cromwell.backend.impl.jes.statuspolling.JesApiQueryManager._
+
+import scala.collection.immutable.Queue
+
+/**
+  * Currently, just holds a set of JES status poll requests until a PollingActor pulls the work.
+  * TODO: Could eventually move all of the JES queries into a single work-pulling model.
+  */
+class JesApiQueryManager extends Actor with ActorLogging {
+
+  private var workQueue: Queue[JesStatusPollQuery] = Queue.empty
+  private var workInProgress: Map[ActorRef, JesPollingWorkBatch] = Map.empty
+
+  // If the statusPoller dies, we want to stop it and handle the termination ourselves.
+  override val supervisorStrategy = SupervisorStrategy.stoppingStrategy
+  private def statusPollerProps = JesPollingActor.props(self)
+  private var statusPoller: ActorRef = _
+  resetStatusPoller()
+
+  override def receive = {
+    case DoPoll(run) => workQueue :+= JesStatusPollQuery(sender, run)
+    case RequestJesPollingWork(maxBatchSize) =>
+      log.debug(s"Request for JES Polling Work received (max batch: $maxBatchSize, current queue size is ${workQueue.size})")
+      handleJesPollingRequest(sender, maxBatchSize)
+    case Terminated(actorRef) => handleTerminated(actorRef)
+    case other => log.error(s"Unexpected message to JesPollingManager: $other")
+  }
+
+  /**
+    * !! Function Impurity Warning: Modifies Actor Data !!
+    */
+  private def handleJesPollingRequest(workPullingJesPollingActor: ActorRef, maxBatchSize: Int) = {
+    workInProgress -= workPullingJesPollingActor
+    val beheaded = beheadWorkQueue(maxBatchSize)
+    beheaded.workToDo match {
+      case Some(work) =>
+        sendWork(workPullingJesPollingActor, JesPollingWorkBatch(work))
+      case None =>
+        log.debug(s"No work for JES poller. Sad.")
+        workPullingJesPollingActor ! NoWorkToDo
+    }
+
+    workQueue = beheaded.newWorkQueue
+  }
+
+  private def sendWork(workPullingJesPollingActor: ActorRef, work: JesPollingWorkBatch) = {
+    log.debug(s"Sending work to JES poller.")
+    workPullingJesPollingActor ! work
+    workInProgress += (workPullingJesPollingActor -> work)
+  }
+
+  private final case class BeheadedWorkQueue(workToDo: Option[NonEmptyList[JesStatusPollQuery]], newWorkQueue: Queue[JesStatusPollQuery])
+  private def beheadWorkQueue(maxBatchSize: Int): BeheadedWorkQueue = {
+
+    val head = workQueue.take(maxBatchSize).toList
+    val tail = workQueue.drop(maxBatchSize)
+
+    head match {
+      case h :: t => BeheadedWorkQueue(Option(NonEmptyList(h, t)), tail)
+      case Nil => BeheadedWorkQueue(None, Queue.empty)
+    }
+  }
+
+  private def handleTerminated(terminee: ActorRef) = {
+    workInProgress.get(terminee) match {
+      case Some(work) =>
```

Comment:
Hmm, probably not. I'll add a message here (I'm also not *too* worried about over-logging this sort of "should **never** happen" situation)